apps/x509.c

```diff
@@ -978,7 +978,7 @@ int x509_main(int argc, char **argv)
                 BIO_printf(bio_err, "Out of memory\n");
                 goto end;
             }
-            BIO_printf(out, "%s Fingerprint=", OBJ_nid2sn(EVP_MD_type(fdig)));
+            BIO_printf(out, "%s Fingerprint=", EVP_MD_name(fdig));
```

Comment:
Same here, use `digestname` (and fall back to `"SHA1"` if it's NULL)

Reply:
Done